source/usage-examples/replaceOne.txt

```diff
@@ -4,31 +4,31 @@ Replace a Document
 
 .. default-domain:: mongodb
 
-You can replace a single document using the :node-api:`replaceOne()
-<Collection.html#replaceOne>` method. ``replaceOne()`` accepts a
-query document and a replacement document, and fully replaces the first
-document that matches the query with the provided replacement document.
-All fields and values in the original document are removed with the
-exception of the ``_id`` value, which remains the same unless you
-explicitly specify a new value for ``_id`` in the replacement document.
+You can replace a single document using the
+:node-api:`collection.replaceOne() <Collection.html#replaceOne>` method.
+``replaceOne()`` accepts a query document and a replacement document. If
+the query matches a document in the collection, it replaces the first
+match with the replacement document. The operation replaces all fields and
+values in the original document with the replacement document. The value of
```

Comment:
This sentence:
```
The operation replaces all fields and
exception of the ``_id`` value, which remains the same unless you
values in the original document with the replacement document.
```

Seems like a regression compared to:
```
 All fields and values in the original document are removed with the 
exception of the ``_id`` value, which remains the same unless you
 explicitly specify a new value for ``_id`` in the replacement document. 
```

Otherwise it is unclear what you mean by "replaces... with the replacement document."

Alternative: you could rephrase like this:
```suggestion
values in the original document with the fields and values in the replacement document. The value of
```

Reply:
The prior version had passive voice ("... are removed...") and was a compound sentence that presented two new concepts. I wanted to break this apart into one part that references only the functionality of replaceOne without the exception case, and then another that explains the exception case.
I think the wording you suggested implies (to me) that there's a mapping between the documents that selectively replaces field-value pairs. I wanted to emphasize that the entire document is replaced (_id field is an exception) rather than just the specified fields. Is there a better wording that you can think of to accomplish this? @nathan-contino-mongo

Reply:
Here's my idea:

```
You can replace a single document using the :node-api:`collection.replaceOne()
<Collection.html#replaceOne>` method. ``replaceOne()`` accepts a
query document and a replacement document. If the query matches a
document in the collection, it replaces the first document that matches
the query with the provided replacement document. This operation
removes all fields and values in the original document and replaces them
with the fields and values in the replacement document. The value of the
``_id`` field remains the same unless you explicitly specify a new value
for ``_id`` in the replacement document.
```

Reply:
I think my biggest issue is the phrase "all fields and values in the original document with the replacement document" because that makes it sound like every field and value becomes a nested document (at least, that's how it sounds to me). This uses all of the grammar/standardization improvements you suggested while keeping the operation's effect explicit.

Reply:
Ok, your suggestion works for me. Thanks for typing it out. I had interpreted your original suggestion slightly differently; it seemed like it only described replacement of fields that existed in the original document which would not be correct.

I guess replacement is a concept that might be interpreted/visualized differently regardless of word choice, so it's good that you've described it with two sentences to reinforce the statement.

Reply:
Thanks! I thought about this a lot when I originally wrote this document so I wanted to be sure that we kept that level of explanation. The differences between update and replace aren't necessarily obvious to a newcomer!